frontend/javascripts/oxalis/model/sagas/isosurface_saga.ts

```diff
@@ -546,17 +615,48 @@ function* loadPrecomputedMeshForSegmentId(
   yield* put(addPrecomputedIsosurfaceAction(layerName, id, seedPosition, meshFileName));
   yield* put(startedLoadingIsosurfaceAction(layerName, id));
   const dataset = yield* select((state) => state.dataset);
+
   let availableChunks = null;
 
+  const availableMeshFiles = yield* call(
+    dispatchMaybeFetchMeshFilesAsync,
+    Store.dispatch,
+    segmentationLayer,
+    dataset,
+    false,
+    false,
+  );
+
+  const meshFile = availableMeshFiles.find((file) => file.meshFileName === meshFileName);
+  if (!meshFile) {
+    Toast.error("Could not load mesh, since the requested mesh file was not found.");
+    return;
+  }
+
+  const version = meshFile.formatVersion;
   try {
-    availableChunks = yield* call(
-      getMeshfileChunksForSegment,
-      dataset.dataStore.url,
-      dataset,
-      getBaseSegmentationName(segmentationLayer),
-      meshFileName,
-      id,
-    );
+    // todo: should actually check against 3, but some new mesh files
+    // still have version 2 ?
+    if (version >= 2) {
+      const segmentInfo = yield* call(
+        meshV3.getMeshfileChunksForSegment,
+        dataset.dataStore.url,
+        dataset,
+        getBaseSegmentationName(segmentationLayer),
+        meshFileName,
+        id,
+      );
+      availableChunks = _.first(segmentInfo.chunks.lods)?.chunks || [];
```

Comment:
Is there a documentation of the new mesh file format somewhere? I would like to know how the `lods` work and whether they are sorted, for example :)

Reply:
See https://www.notion.so/scalableminds/Design-Doc-Mesh-file-069d8fe05fff4017bc88584c1f6f92c6. I think, the lods are not sorted, but since we haven't really done anything multi-lod, yet, I'd defer this until we can properly test it with a multi-lod-file.